### ISSUE TITLE
QA Toolkit - ESC QA Summary for Business Plan Report

### ISSUE THREAD

**Issue by HollySchuelerKCI** [kcigeospatial/sha_oed_issues#370]
Currently, the system is producing an inaccurate output for Q-4 in the Number of quality assurance (QA) inspections performed section of the ESC QA Summary for Business Plan Report.  

The system displays within the table a total numerical value for Q4, while in the bar graph, that numerical value for Q4 does not match the numerical value found in the table.

**Original Request via Email on 11/15/2017**: _The chart on the BP/SS information output have an oddity, see attached. I circled the chart in question , Q-4 should be blank_

![image](https://user-images.githubusercontent.com/25441512/33490483-5a0bad2a-d685-11e7-9b89-cd69f62c2215.png)

**Steps to Replicate:**
1. Click the BP/SS Tab in Level 1
2. Click the Reports Tab in Level 2
3. Locate the ESC QA Summary for Business Plan Reporting Report
4. Enter "2015" for the Fiscal Year
5. Select " All Districts" for the dropdown value.
6. The system will generate the report.
7. In the PDF report, locate the "Number of quality assurace (QA) Inspection Performed" section at the bottom of page1.
8. Notice that the value for Q4 in the table is **5**
9. Notice that the value for Q4 in the bar graph is **1,777**

**Comment by vijme** [comment]
Fixed in dev.

**Comment by HollySchuelerKCI** [comment]
This issue has been successfully validated by the project team in the Development Environment.

**Comment by Tad-Daniel** [comment]
Looks correct

**Comment by HollySchuelerKCI** [comment]
This issue has been confirmed by the project stakeholder in the Development Environment.

This issue has been updated to status = "Ready for Staging"

**Comment by HollySchuelerKCI** [comment]
This issue has been successfully validated by the project team in the Staging Environment.

**Comment by Tad-Daniel** [comment]
It appears to be correct now.  Difficult to tell in the staging env. as there are so few reports.  I am good with this fix.

**Comment by HollySchuelerKCI** [comment]
This issue has been upgraded to status = "Ready for Production"

**Comment by Tad-Daniel** [comment]
The issue has been corrected in Dev and works fine but has caused something else to occur in Prod.  When I try to create a report for 2016 or 2017 everything works fine but when I try to create a report for 2018 I get an unrecoverable error.

**Comment by slmessier** [comment]
@vijme  - could you please investigate the technical details of the system response when generating this report on year 2018?

**Comment by vijme** [comment]
Please find the detailed analysis of this issue. The cause of the issue seems to be:

- There are six orphaned records in the production inspection table that do not have projects associated. 
These inspections cannot be viewed in production as they are not listed anywhere. The most likely case this has happened would be when a user has created a project, added inspections to the project and deleted the project. The project has got deleted in the table while the inspections are still in the system.  

- Unless there could be any specific reason why the inspections have not been deleted when the project was deleted, (like, for example, they still needed to be on the inspections count), we can correct this issue in production by deleting the existing orphaned inspections manually, which will temporarily fix the issue in production and deleting the associated inspections for a project during project deletion in future.  **This is the recommended solution**

- If the inspections are still needed, then we can fix the issue by  modifying the query for this specific issue. **This is the alternate solution**

_Basically, I would like to know whether we need the inspections for the deleted projects still to be in the system._

**Comment by Tad-Daniel** [comment]
Can you list the six reports for me.  I am thinking the recommended solution but need to verify the reports first.  

Just to confirm - when a project is deleted the associated data remains in the system?

**Comment by slmessier** [comment]
@vijme could you please provide a list of the 6 inspection records that are not associated with any project?

**Comment by vijme** [comment]
Just to confirm - when a project is deleted the associated data remains in the system?
YES 


![image](https://user-images.githubusercontent.com/6668314/36209399-9f6abf5e-1169-11e8-8e5b-13929696db7f.png)

**Comment by Tad-Daniel** [comment]
Some of these orphaned records appear to be from the project I recently deleted by accident that you are also looking into.  I would prefer not to delete any until we resolve that issue so maybe the alternate solution is the best (modify the query).  This would also stop the issue from reoccurring.

![image](https://user-images.githubusercontent.com/7913678/36216087-cb6f8a32-117b-11e8-96b0-9a99406b61c7.png)

![image](https://user-images.githubusercontent.com/7913678/36215786-0a51db66-117b-11e8-9946-41993f47326c.png)

**Comment by slmessier** [comment]
@vijme are you able to move forward with development by modifying the supporting query?

**Comment by vijme** [comment]
Query modified and tested by deleting projects with final inspections.. I hope it fixes the issue in production.

**Comment by Tad-Daniel** [comment]
It appears to be working properly in DEV

**Comment by HollySchuelerKCI** [comment]
This issue has been successfully validated by the project team and confirmed by stakeholder in the Development Environment.

This issue has been updated to status = "Ready for Staging"

**Comment by HollySchuelerKCI** [comment]
This issue has been successfully validated by the project team in the Staging Environment

**Comment by Tad-Daniel** [comment]
Looks good

**Comment by HollySchuelerKCI** [comment]
This issue has been confirmed by stakeholders in the Staging Environment.

This issue has been updated to status = "Ready for Production"

**Comment by Tad-Daniel** [comment]
Appears correct now. item complete

**Comment by HollySchuelerKCI** [comment]
The status of this issue = Closed
